SampleApp/RoutingViewController.swift

```diff
@@ -41,11 +41,23 @@ class RoutingViewController: UIViewController, RoutingSearchDelegate {
     }
   }
 
+  override func viewDidAppear(animated: Bool) {
+    super.viewDidAppear(animated)
+    if LocationManager.sharedManager.isAlwaysAuthorized() || LocationManager.sharedManager.isInUseAuthorized() {
+      LocationManager.sharedManager.startUpdatingLocation()
```

Comment:
I went this way because it's the way I would except people to implement it in an app, if that makes sense? 

The problem with requesting location in a "one off" scenario is the location may not be current, or core location / our location manager may not have one at all, so we would have to heat up the modem/GPS to get a fix and that could take a bit, meanwhile the user is waiting. Doing it this way, the `LocationManager`'s `.currentLocation` property is getting updated and caching the last known good location, which we're then using on  [line 58](https://github.com/mapzen/ios/pull/135/files/6a963c975a01eb8b4225fe22cee38f14776d2267#diff-5e8f16b3beb0c57a7af33e8a0c0c53b8R58) when someone makes a request to route.

Obviously it's a little overkill for a sample app but I think it provides the best UX.